TextFieldEffects/TextFieldEffects/HoshiTextField.swift

```diff
@@ -134,12 +134,34 @@ typealias VoidClosure = () -> Void
     configureErrorLabel()
     updateBorder()
     updatePlaceholder()
+    addSublayers()
+    addLabels()
+    frame = CGRect(x: frame.origin.x,
+                   y: frame.origin.y,
+                   width: frame.size.width,
+                   height: height)
+  }
+
+  private func addSublayers() {
     layer.addSublayer(inactiveBorderLayer)
     layer.addSublayer(activeBorderLayer)
+  }
+
+  private func addLabels() {
     addSubview(placeholderLabel)
     addSubview(errorLabel)
   }
 
+  open override func setNeedsLayout() {
```

Comment:
qq: I wander if overriding `layoutSubviews()` would do the same, but is recommended way per Apple docs?

Reply:
it's called less often, that's why I moved it there, and also it was a little bit more intuitive to be when the methods was going to be called, `drawViewsForRect` didn't give me a hint on when it was going to be called.
Thoughts?

Reply:
@Sh4rky As we talked today, I agree that this seems to be the optimal solution at this time.